### PR TITLE
Fix UIWindowSceneGeometryPreferencesIOS leak

### PR DESCRIPTION
**Pull request by jmagman**
Fixes https://github.com/flutter/flutter/issues/124288

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [Flutter Style Guide] and the [C++, Objective-C, Java style guides].
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I added new tests to check the change I am making or feature I am adding, or Hixie said the PR is test-exempt. See [testing the engine] for instructions on writing and running engine tests.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I signed the [CLA].
- [x] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[C++, Objective-C, Java style guides]: https://github.com/flutter/engine/blob/main/CONTRIBUTING.md#style
[testing the engine]: https://github.com/flutter/flutter/wiki/Testing-the-engine
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat

### CODE REVIEW

File: shell/platform/darwin/ios/framework/Source/FlutterViewControllerTest.mm [hunk 1]
```diff
@@ -1341,47 +1341,53 @@ - (void)orientationTestWithOrientationUpdate:(UIInterfaceOrientationMask)mask
...
-  FlutterViewController* realVC = [[FlutterViewController alloc] initWithEngine:self.mockEngine
-                                                                        nibName:nil
-                                                                         bundle:nil];
-  if (@available(iOS 16.0, *)) {
-    mockWindowScene = OCMClassMock([UIWindowScene class]);
-    if (realVC.supportedInterfaceOrientations == mask) {
-      OCMReject([mockWindowScene requestGeometryUpdateWithPreferences:[OCMArg any]
-                                                         errorHandler:[OCMArg any]]);
+  __block __weak id weakPreferences;
+  @autoreleasepool {
```
Comment (jmagman) [inline shell/platform/darwin/ios/framework/Source/FlutterViewControllerTest.mm:1345]: Most of this change is whitespace, this is new.

File: shell/platform/darwin/ios/framework/Source/FlutterViewControllerTest.mm [hunk 2]
```diff
@@ -1341,47 +1341,53 @@ - (void)orientationTestWithOrientationUpdate:(UIInterfaceOrientationMask)mask
...
-  [realVC performOrientationUpdate:mask];
-  if (@available(iOS 16.0, *)) {
-    OCMVerifyAll(mockWindowScene);
-  } else {
-    OCMVerifyAll(deviceMock);
   }
+  [mockWindowScene stopMocking];
   [deviceMock stopMocking];
   [mockApplication stopMocking];
+  XCTAssertNil(weakPreferences);
```
Comment (jmagman) [inline shell/platform/darwin/ios/framework/Source/FlutterViewControllerTest.mm:1390]: And these two line changes are new.

**Review by hellohuanlin — APPROVED** [review]
thanks for cleaning this up

**Review by cyanglaz — APPROVED** [review]
LGTM! Thanks!

**Comment by jmagman** [comment]
I don't know why `flutter-gold` is even running for this commit, nor why it's stuck pending.  Filed https://github.com/flutter/flutter/issues/124295.

**Pull request merged**